### PR TITLE
Adds support for review missed logic #8607

### LINKED ISSUE #8607: Review Missed
**Is your feature request related to a problem? Please describe.**
CARE has a feature where healthcare workers can set review time. And when no log update is filed for the patient within the time, a "Review Missed" label appears on the patient card in the patient list (https://care-staging.ohc.network/patients?page=1&limit=12&review_missed=true) and the Patient dashboard (https://care-staging.ohc.network/facility/25059be9-968b-4e30-9b64-4c6ecc35ea98/patient/b4d6ccda-685e-4b92-95e1-e5b10f36f378/consultation/20bd5ddc-3ecb-4c2f-92e7-0d2b37ef2ecd)

The patient dashboard also shows the date and time when the review was supposed to happen along with the review missed label. 

The patient list has a filter by which user can filter and find the list of patients who have missed the review.

The patient list may also be sorted by review missed date. 

However, the doctors find it difficult to understand from the list if such a review was missed by a day/week/month. This will also determine the urgency with which the doctor needs to review the patient.

**Describe the solution you'd like**
The solution must enable user to understand, in relative terms, when was the review missed. And allow the doctor to list patients based on duration since review is missed. 

Any feature to this effect is needed.

A few of the solutions I recommend are:

- [ ] On patient list, show relative time since review missed ("review missed- since 26 Days")
- [ ] In advanced filter, review missed filter to have the following options: "No, missed by < 1 day, missed by 1-7 days, missed by 7-14 days, missed by 14-28 days, missed by >28 days) 

(@athulmanuel please comment if the filter is needed. Would adding the relative time alone fix the problem?)

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Sancharisingh**
## Proposed Changes

- Fixes #8607 
- Added success and purple variants in badge.tsx.
- Implemented "Review Missed" logic in managePatients.tsx.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [X] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Badge` component for patient status notifications, replacing the previous `Chip` component.
	- Added new badge variants, including "primary" for upcoming reviews and "destructive" for missed reviews, enhancing visual clarity.
	- Expanded badge size options to include small, medium, and large.
	- Enhanced `Badge` component functionality with optional properties for icons and tooltips.

- **Bug Fixes**
	- Improved clarity of patient status indicators by updating to the `Badge` component for various notifications.

### CODE REVIEW

**Comment by Jacobjeevan** [comment]
Try to salvage the other PR instead #9259 

Since you haven't force pushed, you should be able to rollback all the changes (just push your new code on stash beforehand) and then merge all the changes since you opened the PR/merge and pop from stash.

**Comment by Sancharisingh** [comment]
Hi @Jacobjeevan @rithviknishad ,

Thank you for your guidance! I have made the changes you suggested. Could you please reopen PR #9259 so I can push the updated changes?

Looking forward to your response.

Best regards,
Sanchari singh

**Comment by Jacobjeevan** [comment]
> Hi @Jacobjeevan @rithviknishad ,
> 
> Thank you for your guidance! I have made the changes you suggested. Could you please reopen PR #9259 so I can push the updated changes?
> 
> Looking forward to your response.
> 
> Best regards, Sanchari singh

You should have the option to reopen it 👍  But looks like the branch is deleted, so let's use this PR instead then.

File: src/components/Patient/ManagePatients.tsx [hunk 1]
```diff
@@ -589,24 +591,26 @@ export const PatientManager = () => {
...
-                        size="small"
-                        variant="danger"
-                        startIcon="l-clock"
-                        text="Review Missed"
-                      />
+                    Number(patient.last_consultation?.review_interval) > 0 && (
+                      <Badge
+                        variant={
+                          dayjs().isAfter(patient.review_time)
+                            ? "purple"
```
Comment (Jacobjeevan) [inline src/components/Patient/ManagePatients.tsx]: Let's use the destructive instead 🤔 That should be in line in what we had originally (red color).

Primary should work for the green color.

Try both of those and remove success/purple styles from badge component if they work.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Patient/ManagePatients.tsx (hunk 1 above)
Comment (Sancharisingh) [reply src/components/Patient/ManagePatients.tsx]: Ok

Comment (Sancharisingh) [reply src/components/Patient/ManagePatients.tsx]: There is no primary, so I changed the name of 'success' to 'primary.'

Comment (Jacobjeevan) [reply src/components/Patient/ManagePatients.tsx]: image

Okay, let's modify the primary to match the chip component's primary style.

While you are at it, can you also update the existing chips in the page to use the new badge component?

File: src/components/ui/badge.tsx [hunk 2]
```diff
@@ -17,6 +17,8 @@ const badgeVariants = cva(
         warning:
           "border-transparent bg-yellow-400 text-gray-900 shadow hover:bg-yellow-500 dark:bg-yellow-400 dark:text-gray-900 dark:hover:bg-yellow-500",
         outline: "text-gray-950 dark:text-gray-50",
+        primary:
+          "border-transparent bg-green-500 text-gray-50 shadow hover:bg-green-500/80 dark:bg-green-900 dark:text-gray-50 dark:hover:bg-green-900/80",
```
Comment (rithviknishad) [inline src/components/ui/badge.tsx:22]: primary should use primary :) 
```suggestion
          "border-transparent bg-primary-500 text-gray-50 shadow hover:bg-primary-500/80 dark:bg-primary-900 dark:text-gray-50 dark:hover:bg-primary-900/80",
```

File: src/components/Patient/ManagePatients.tsx (hunk 1 above)
Comment (Jacobjeevan) [reply src/components/Patient/ManagePatients.tsx]: @Sancharisingh Can you do the above as well (switch existing chips in the component to use badge)?

Comment (Sancharisingh) [reply src/components/Patient/ManagePatients.tsx]: yes

Comment (Sancharisingh) [reply src/components/Patient/ManagePatients.tsx]: Hi @Jacobjeevan,
I updated the Chip to Badge in the ManagePatient file. Should this change be applied project-wide, or is it fine to keep other instances as they are?

Comment (Jacobjeevan) [reply src/components/Patient/ManagePatients.tsx]: > Hi @Jacobjeevan, I updated the Chip to Badge in the ManagePatient file. Should this change be applied project-wide, or is it fine to keep other instances as they are?

Just this file/component is enough.

**Comment by Jacobjeevan** [comment]
image


Seeing all these together 😅 and all the consultations filed ones, I feel these might be a bit too intense. Let's ping @vinutv 
and see what he thinks. If we are keeping current styles (below), we can either add or modify the existing variants.

image

File: src/components/Patient/ManagePatients.tsx (hunk 1 above)
Comment (Sancharisingh) [reply src/components/Patient/ManagePatients.tsx]: Okay.

**Comment by Sancharisingh** [comment]
Yeah, I am also thinking of modifying all these variants.

**Comment by AdityaJ2305** [comment]
@nihal467  @vinutv any updates on this ?

**Comment by Sancharisingh** [comment]
Hi @rithviknishad @vinutv @Jacobjeevan,
Are there any updates?

**Comment by Jacobjeevan** [comment]
> Hi @rithviknishad @vinutv @Jacobjeevan, Are there any updates?

Closing as not relevant in new architecture.

**Pull request closed without merging**